internal/daemon/controller/handlers/authmethods/authmethod_service.go

```diff
@@ -78,6 +78,11 @@ var (
 		resource.Account:      accounts.CollectionActions,
 		resource.ManagedGroup: managed_groups.CollectionActions,
 	}
+
+	additionalResourceGrants = []resource.Type{
+		resource.Account,
+		resource.ManagedGroup,
+	}
```

Comment:
Rather than having these lists defined in each service handler, it might be nice to define these as a method on `resource.Type` so that the grouping of parent and child types are all defined with the types. We already have something related to this with the [`Parent()` and `HasChildTypes()` methods](https://github.com/hashicorp/boundary/blob/f91978747b3266bf7d989d1e04682e374bbb76e3/internal/types/resource/resource.go#L144-L166)

With that, we wouldn't need the `WithFetchAdditionalResourceGrants` option, since we could just call that method on the passed in resource when fetching grants.